generate/state.go

```diff
@@ -372,6 +376,59 @@ func fixEdges(g *graph.Graph, cfg map[string]map[string]interface{}, opt Options
 					}
 				}
 			}
+
+		}
+	}
+
+	for id, nodes := range imNodes {
+		edges := g.GetEdgesForNode(id)
+		for _, e := range edges {
+			// We only want edges that have the
+			// current Node (that is Edge) as Target
+			if e.Target != id {
+				continue
+			}
+			for _, no := range nodes {
+				pv, _, err := getProviderAndResource(no, opt)
+				if err != nil {
+					return err
+				}
+
+				res, err := pv.Resource(no)
```

Comment:
I think the naming could be improved here.

I assumed by reading it that `getProviderAndResource` would already return a `*resource.Resource` as second argument, while it seems to simply be the `Name` of the Resource, which is not what we want - thus the need of calling `pv.Resource` afterwards.

Why not returning the full Resource? Also, why are `IsNode` / `IsEdge` have a Provider as a receiver rather than a `Resource` wouldn't it make more sense?

Reply:
This are not changes of this PR but ok haha (meaning that it's just used logic not defined).

~First of all there is no such thing as `*resource.Resource` on IM so that would not be an issue.~ Ok you mean a `tfdocs/resource.Resource`. The easy answer is that we do not want the `resource.Resource` in most of the cases.

`getProviderAndResource`: With param `aws_instance.front` would return `aws.Provider, "aws_instance"`

`IsNode/IsEdge`: It's on the provider as it's the one that has the logic (it's on `providers/aws/graph.go`) and the `resource.Rresource` is from TFDocs and it's more related to a Resource documentation.

Basically the `tfdocs/resource.Resource` only has information of the Documentation of the resource, it has a `Metadata map[string]struct{}` that we could push a `"node": true|false` and add a function to that repositry to check that, but I feel it'll create an inderect connection between the 2 libs `tfdcos` and `inframap` which I would try to avoid and keep all the IM logic inside IM.